mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -236,6 +237,45 @@ def _startup(self, bus, skill_id=""):
                 pass
             raise e
 
+    def _init_settings(self):
+        """Setup skill settings."""
+        LOG.debug(f"initializing skill settings for {self.skill_id}")
+
+        # if settings were used in __init__ method self._startup won't have been called yet
+        # assume these are default values
+        if self.settings:
+            self._initial_settings = copy(self.settings)
+
+        # migrate settings if needed
+        if not exists(self.settings_path) and exists(self._old_settings_path):
+            shutil.copy(self._old_settings_path, self.settings_path)
+
+        # NOTE: lock is disabled due to usage of deepcopy and to allow json serialization
+        self.settings = JsonStorage(self.settings_path, disable_lock=True)
+
+        # copy default values set in __init__
+        for k, v in self._initial_settings.items():
+            if k not in self.settings:
+                self.settings[k] = v
+        self._initial_settings = copy(self.settings)
+
+    @property
+    def _old_settings_path(self):
+        old_dir = self.config_core.get("data_dir", "/opt/mycroft")
+        old_folder = self.config_core.get("skills", {}).get("msm", {}).get("directory", "skills")
+        return join(old_dir, old_folder, self.skill_id, 'settings.json')
+
+    @property
+    def settings_path(self):
+        is_xdg = is_using_xdg()
+        if self.settings_write_path:
+            if is_xdg:
+                LOG.warning("self.settings_write_path has been deprecated!")
```

Comment:
I think this is valid to log in any case; the correct method would be to set the xdg param anyways instead of settings path directly